frame/vesting/src/benchmarking.rs

```diff
@@ -160,68 +178,247 @@ benchmarks! {
-			Some(T::MinVestedTransfer::get()),
+			Some(transfer_amount),
 			"Lock not created",
 		);
 	}
 
-	force_vested_transfer {
+	first_force_vested_transfer {
 		let l in 0 .. MaxLocksOf::<T>::get();
 
 		let source: T::AccountId = account("source", 0, SEED);
 		let source_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(source.clone());
 		T::Currency::make_free_balance_be(&source, BalanceOf::<T>::max_value());
+
 		let target: T::AccountId = account("target", 0, SEED);
 		let target_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(target.clone());
 		// Give target existing locks
 		add_locks::<T>(&target, l as u8);
 
-		let transfer_amount = T::MinVestedTransfer::get();
+		let duration = 10u32;
+		let transfer_amount = (ED * duration).into();
 
-		let vesting_schedule = VestingInfo {
-			locked: transfer_amount,
-			per_block: 10u32.into(),
-			starting_block: 1u32.into(),
-		};
-	}: _(RawOrigin::Root, source_lookup, target_lookup, vesting_schedule)
+		let vesting_schedule = VestingInfo::try_new::<T>(
+			transfer_amount,
+			duration.into(),
+			1u32.into(),
+		)?;
+	}: force_vested_transfer(RawOrigin::Root, source_lookup, target_lookup, vesting_schedule)
 	verify {
 		assert_eq!(
-			T::MinVestedTransfer::get(),
+			transfer_amount,
 			T::Currency::free_balance(&target),
 			"Transfer didn't happen",
 		);
 		assert_eq!(
 			Vesting::<T>::vesting_balance(&target),
-			Some(T::MinVestedTransfer::get()),
+			Some(transfer_amount),
 			"Lock not created",
 		);
 	}
+
+	last_force_vested_transfer {
+		let l in 0 .. MaxLocksOf::<T>::get();
+
+		let source: T::AccountId = account("source", 0, SEED);
+		let source_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(source.clone());
+		T::Currency::make_free_balance_be(&source, BalanceOf::<T>::max_value());
+
+		let target: T::AccountId = account("target", 0, SEED);
+		let target_lookup: <T::Lookup as StaticLookup>::Source = T::Lookup::unlookup(target.clone());
+		// Give target existing locks
+		add_locks::<T>(&target, l as u8);
+		// Add one less than max vesting schedules
+		let one_less_than_max_vesting = T::MaxVestingSchedules::get() - 1;
+		let mut expected_balance = add_vesting_schedules::<T>(target_lookup.clone(), one_less_than_max_vesting)?;
+
+		let duration = 10u32;
+		let transfer_amount = (ED * duration).into();
+		expected_balance += transfer_amount;
+
+		let vesting_schedule = VestingInfo::try_new::<T>(
+			transfer_amount,
+			duration.into(),
+			1u32.into(),
+		)?;
+	}: force_vested_transfer(RawOrigin::Root, source_lookup, target_lookup, vesting_schedule)
+	verify {
+		assert_eq!(
+			expected_balance,
+			T::Currency::free_balance(&target),
+			"Transfer didn't happen",
+		);
+		assert_eq!(
+			Vesting::<T>::vesting_balance(&target),
+			Some(expected_balance.into()),
+				"Lock not correctly updated",
+			);
+		}
+
+	not_unlocking_merge_schedules {
+		let l in 0 .. MaxLocksOf::<T>::get();
```

Comment:
would it be helpful if i also did this but for `T::MaxVestingSchedules::get()`?

i don't quite understand if it would improve the quality of the benchmark

Reply:
If the complexity of the operation is noticeably different in the last one, maybe. 

You also have the option to just always assume worse-case, if benchmarking it accurately is too much of a hassle.